src/components/text-editor/prosemirror-adapter/plugins/menu-state-tracking-plugin.ts

```diff
@@ -7,24 +7,31 @@ import { EditorMenuTypes } from '../menu/types';
 
 export const actionBarPluginKey = new PluginKey('actionBarPlugin');
 
-export type ActiveMenuItems = Record<EditorMenuTypes, boolean>;
+export interface ActiveMenuItems {
+    active: Record<EditorMenuTypes, boolean>;
+    allowed: Record<EditorMenuTypes, boolean>;
+}
 
-export type UpdateMenuItemsCallBack = (activeTypes: ActiveMenuItems) => void;
+export type UpdateMenuItemsCallBack = (
+    activeTypes: Record<EditorMenuTypes, boolean>,
+    allowedTypes: Record<EditorMenuTypes, boolean>,
+) => void;
 
-const getMenuItemStates = (
+export const getMenuItemStates = (
     menuTypes: EditorMenuTypes[],
     menuCommandFactory: MenuCommandFactory,
     view: EditorView,
 ): ActiveMenuItems => {
-    const activeTypes: ActiveMenuItems = {};
+    const activeTypes: Record<EditorMenuTypes, boolean> = {};
+    const allowedTypes: Record<EditorMenuTypes, boolean> = {};
 
     menuTypes.forEach((type) => {
         const command: CommandWithActive = menuCommandFactory.getCommand(type);
-        activeTypes[type] =
-            command && command.active && command.active(view.state);
+        activeTypes[type] = command?.active?.(view.state) || false;
```

Comment:
```suggestion
        activeTypes[type] = !!command?.active?.(view.state) || false;
```

This way, if `command` or `active` is null or undefined, we still set the value to `false`, rather than `null` or `undefined`.